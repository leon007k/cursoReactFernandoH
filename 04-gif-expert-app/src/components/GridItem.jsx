import { PropTypes } from 'prop-types'
import '../style.css'

export function GridItem({ title, url }) {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  )
}

GridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
}