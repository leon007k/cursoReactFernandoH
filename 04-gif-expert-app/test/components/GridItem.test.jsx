import { render,screen } from "@testing-library/react";
import { GridItem } from "../../src/components/GridItem.jsx";

/* eslint-disable no-undef */
describe('Pruebas del componente <GridItem />', () => {
  const title = 'solo leveling';
  const url = 'https://media.giphy.com/media/3o7TKz9bX9Nv8a0lHa/giphy.gif';


  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GridItem title={title} url={url} />)
    expect(container).toMatchSnapshot();
  });


  test('Debe mostrar la imagen con el URL y el ALT indicado', () => {
    render(<GridItem title={title} url={url} />);
    // * screen.debug(); --> Nos permite ver el HTML que se estÃ¡ renderizando
    // * screen.getByRole('img') --> Nos permite seleccionar un elemento por su rol, en este caso una imagen
    // * y esperamos que el src de la imagen sea igual al url que le pasamos al igual que el alt
    // expect(screen.getByRole('img').src).toBe(url);
    const {src, alt} = screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);
  })

  test('Debe de mostrar el titulo en el componente', () =>{
    render(<GridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  })
});