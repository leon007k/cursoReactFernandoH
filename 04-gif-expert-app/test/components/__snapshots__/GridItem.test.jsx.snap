// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GridItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="solo leveling"
      src="https://media.giphy.com/media/3o7TKz9bX9Nv8a0lHa/giphy.gif"
    />
    <p>
      solo leveling
    </p>
  </div>
</div>
`;
